src/Symfony/Component/Scheduler/Messenger/SchedulerTransport.php

```diff
@@ -23,13 +23,14 @@ class SchedulerTransport implements TransportInterface
 {
     public function __construct(
         private readonly MessageGeneratorInterface $messageGenerator,
+        private readonly string $name,
     ) {
     }
 
     public function get(): iterable
     {
         foreach ($this->messageGenerator->getMessages() as $message) {
-            yield Envelope::wrap($message, [new ScheduledStamp()]);
+            yield Envelope::wrap($message, [new ScheduledStamp($this->name)]);
```

Comment:
but that doesn't work: the name is something that belongs to a schedule, not to the transport. The generated messages can come from many schedules.

Reply:
Each schedule has its own transport though.